design-proposals/sidecar-hooks.md

```diff
@@ -0,0 +1,56 @@
+# Sidecar hooks
+
+## Overview
+
+A hooking mechanism that allows customizations to the VMI before it starts without changing KubeVirt
+core components. The common user of those hooks are Sidecars that will run in the same Pod that the
+VMI will be deployed.
+
+## Motivation
+
+A way to accommodate use cases that we might not want to bring and support in KubeVirt itself or to
+give room to test and develop features while they are still not mature enough.
+
+## Goals
+
+- Provide and maintain a set of gRPC APIs for users to do modifications to the VMI
+- Provide and maintain sidecars that are important for core functionalities
```

Comment:
I am unsure if this merits a special goal. Both external and internal sidecars are expected to get the same assurance and stability for the integration point.

Reply:
Right, what you said is true but should be covered by the first bullet. What the second bullet introduces is that, we will provide some sidecars which add functionalities to KubeVirt and those sidecars we will be maintaining.

We might want to take out the second bullet out or not. Not sure where to draw the line in this document.